lib/Jsrt/JsrtDebugManager.cpp

```diff
@@ -0,0 +1,693 @@
+            JsrtDebugUtils::AddPropertyToObject(eventDataObject, JsrtDebugPropertyId::error, compileException->ei.bstrDescription, ::SysStringLen(compileException->ei.bstrDescription), scriptContext);
+            JsrtDebugUtils::AddPropertyToObject(eventDataObject, JsrtDebugPropertyId::line, compileException->line, scriptContext);
+            JsrtDebugUtils::AddPropertyToObject(eventDataObject, JsrtDebugPropertyId::column, compileException->ichMin - compileException->ichMinLine - 1, scriptContext); // Converted to 0-based
+            JsrtDebugUtils::AddPropertyToObject(eventDataObject, JsrtDebugPropertyId::sourceText, compileException->bstrLine, ::SysStringLen(compileException->bstrLine), scriptContext);
+        }
+        else
+        {
+            JsrtDebugDocumentManager* debugDocumentManager = this->GetDebugDocumentManager();
+            Assert(debugDocumentManager != nullptr);
+
+            // Create DebugDocument and then report JsDiagDebugEventSourceCompile event
+            Js::DebugDocument* debugDocument = HeapNewNoThrow(Js::DebugDocument, utf8SourceInfo, scriptFunction->GetFunctionBody());
+            if (debugDocument != nullptr)
+            {
+                utf8SourceInfo->SetDebugDocument(debugDocument);
+
+                // Only add scriptId if everything is ok as scriptId is used for other operations
+                JsrtDebugUtils::AddScriptIdToObject(eventDataObject, utf8SourceInfo);
+            }
+            jsDiagDebugEvent = JsDiagDebugEventSourceCompile;
+        }
+
+        this->CallDebugEventCallback(jsDiagDebugEvent, eventDataObject, scriptContext);
+    }
+}
+
+void JsrtDebugManager::ReportBreak(Js::InterpreterHaltState* haltState)
+{
+    if (this->debugEventCallback != nullptr)
+    {
+        Js::FunctionBody* functionBody = haltState->GetFunction();
+        Assert(functionBody != nullptr);
+
+        Js::Utf8SourceInfo* utf8SourceInfo = functionBody->GetUtf8SourceInfo();
+        int currentByteCodeOffset = haltState->GetCurrentOffset();
+        Js::ScriptContext* scriptContext = utf8SourceInfo->GetScriptContext();
+
+        JsDiagDebugEvent jsDiagDebugEvent = this->GetDebugEventFromStopType(haltState->stopType);
+
+        JsrtDebugEventObject debugEventObject(scriptContext);
+
+        Js::DynamicObject* eventDataObject = debugEventObject.GetEventDataObject();
+
+        Js::ProbeContainer* probeContainer = scriptContext->GetDebugContext()->GetProbeContainer();
+
+        if (jsDiagDebugEvent == JsDiagDebugEventBreak)
+        {
+            UINT bpId = 0;
+            probeContainer->MapProbesUntil([&](int i, Js::Probe* pProbe)
+            {
+                Js::BreakpointProbe* bp = (Js::BreakpointProbe*)pProbe;
+                if (bp->Matches(functionBody, utf8SourceInfo->GetDebugDocument(), currentByteCodeOffset))
+                {
+                    bpId = bp->GetId();
+                    return true;
+                }
+                return false;
+            });
```

Comment:
Would be a good if this whole thing move to probecontainer - which will talk to breakpointprobe and give id. No need to expose that logic here.